TYPO3.Media/Classes/TYPO3/Media/Domain/Service/ThumbnailService.php

```diff
@@ -204,13 +192,11 @@ public function getUriForThumbnail(Thumbnail $thumbnail)
             ), 1450178437);
         }
 
-        if (preg_match('#^resource://([^/]+)/Public/(.*)#', $staticResource, $matches) !== 1) {
-            throw new ThumbnailServiceException(sprintf(
-                'Invalid static resource path "%s" for static thumbnail "%s".',
-                $staticResource,
-                $this->persistenceManager->getIdentifierByObject($thumbnail)
-            ), 1450188242);
+        try {
+            list($package, $path) = $this->resourceManager->getPackageAndPathByPublicPath($staticResource);
+            return $this->resourceManager->getPublicPackageResourceUri($package, $path);
+        } catch (Exception $exception) {
+            return $staticResource;
```

Comment:
This changes the logic to return something where an exception would have been thrown before. What effect does that have? Especially since it now returns something that was considered broken in the previous implementation.